src/kubeobjects/object.go

```diff
@@ -1,9 +1,21 @@
 package kubeobjects
 
-import "sigs.k8s.io/controller-runtime/pkg/client"
+import (
+	"context"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
 func Key(object client.Object) client.ObjectKey {
 	return client.ObjectKey{
 		Name: object.GetName(), Namespace: object.GetNamespace(),
 	}
 }
+
+func EnsureDeleted(ctx context.Context, client client.Client, obj client.Object) error {
+	if err := client.Delete(ctx, obj); err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
```

Comment:
```suggestion
func Delete(ctx context.Context, client client.Client, obj client.Object) error {
    err := client.Delete(ctx, obj)
    
	if err != nil && !k8serrors.IsNotFound(err) {
		return err
	}
	
	return nil
}
```
Or `SafeDelete`, but `EnsureDeleted` sounds like "I have deleted something in a previous step, now I call a function that tells me if it really has been deleted or not"